src/main/java/uk/gov/justice/digital/client/s3/S3DataProvider.java

```diff
@@ -80,12 +80,23 @@ public Dataset<Row> getStreamingSourceDataWithSchemaInference(SparkSession spark
         }
     }
 
-    public Dataset<Row> getBatchSourceData(SparkSession sparkSession, List<String> filePaths) throws SparkException {
-        val scalaFilePaths = JavaConverters.asScalaIteratorConverter(filePaths.iterator()).asScala().toSeq();
-        return sparkSession
-                .read()
-                .option("mergeSchema", "true")
-                .parquet(scalaFilePaths);
+    public Dataset<Row> getBatchSourceData(SparkSession sparkSession, List<String> filePaths) throws DataProviderFailedMergingSchemasException {
+        try {
+            val scalaFilePaths = JavaConverters.asScalaIteratorConverter(filePaths.iterator()).asScala().toSeq();
+            return sparkSession
+                    .read()
+                    .option("mergeSchema", "true")
+                    .parquet(scalaFilePaths);
+        } catch (Exception e) {
+            // We can't catch the specific SparkException type we actually want here because scala erases the fact
```

Comment:
I thought moving this scala vs java nastiness in here and throwing our own exception would keep the main part of the app a bit cleaner